commit 76a0ce2b847924990a5b285bdcc0185c3c6dae60 (refs/changes/78/1478/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-10T15:21:29-08:00 (2 years, 8 months ago)
    
    joyent/node-cueball#93 kang payload missing "options.domain" etc
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/pool-monitor.js b/lib/pool-monitor.js
index 51718fe..a0fc413 100644
--- a/lib/pool-monitor.js
+++ b/lib/pool-monitor.js
@@ -102,9 +102,14 @@ CueBallPoolMonitor.prototype.toKangOptions = function () {
 		obj.state = pool.getState();
 		obj.counters = pool.p_counters;
 		obj.options = {};
-		obj.options.domain = pool.p_resolver.r_domain;
-		obj.options.service = pool.p_resolver.r_service;
-		obj.options.defaultPort = pool.p_resolver.r_defport;
+		var res = cset.cs_resolver;
+		mod_assert.object(res);
+		mod_assert.object(res.r_fsm);
+		res = res.r_fsm;
+		obj.options.domain = res.r_domain || pool.p_domain;
+		mod_assert.string(obj.options.domain);
+		obj.options.service = res.r_service;
+		obj.options.defaultPort = res.r_defport;
 		obj.options.spares = pool.p_spares;
 		obj.options.maximum = pool.p_max;
 		return (obj);
@@ -144,9 +149,14 @@ CueBallPoolMonitor.prototype.toKangOptions = function () {
 		obj.target = cset.cs_target;
 		obj.maximum = cset.cs_max;
 		obj.options = {};
-		obj.options.domain = cset.cs_resolver.r_domain;
-		obj.options.service = cset.cs_resolver.r_service;
-		obj.options.defaultPort = cset.cs_resolver.r_defport;
+		var res = cset.cs_resolver;
+		mod_assert.object(res);
+		mod_assert.object(res.r_fsm);
+		res = res.r_fsm;
+		obj.options.domain = res.r_domain || cset.cs_domain;
+		mod_assert.string(obj.options.domain);
+		obj.options.service = res.r_service;
+		obj.options.defaultPort = res.r_defport;
 		return (obj);
 	}
 
